File: backend/src/Locker.Backend.Domain/Entities/Locker.cs
namespace Locker.Backend.Domain.Entities;

public class Locker : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public bool IsAutoLockEnabled { get; set; } = true;
    public bool IsIntrusionAlertEnabled { get; set; } = true;
    public List<LockerSlot> Slots { get; set; } = new();
    public bool IsDeleted { get; set; } = false;
}

File: mobile/lib/features/locker/domain/entities/locker.dart
import 'package:equatable/equatable.dart';

import 'locker_slot.dart';

/// Pure domain entity - không phụ thuộc framework hay data layer
class Locker extends Equatable {
  final String id;
  final String name;
  final String location;
  final double latitude;
  final double longitude;
  final List<LockerSlot> slots;

  const Locker({
    required this.id,
    required this.name,
    this.location = '',
    this.latitude = 0.0,
    this.longitude = 0.0,
    this.slots = const [],
  });

  // Dùng cho trường hợp orElse trong UI
  static const Locker empty = Locker(
    id: '',
    name: 'N/A',
    location: '',
    latitude: 0.0,
    longitude: 0.0,
    slots: [],
  );

  @override
  List<Object?> get props => [id, name, location, latitude, longitude, slots];
}
